### PR TITLE
Enforce stricter hasPendingUpdate check

### PR DESCRIPTION
**Pull request by quanganhdo**
Task/Issue URL: https://app.asana.com/0/1201048563534612/1208746324713624/f
Tech Design URL:
CC:

**Description**:

Fixes Install Now menu item does nothing by making sure we can resume the update process. As a bonus, this fixes an edge case where the user toggles back and forth between automatic & manual update checkboxes and that causes the upgrade process to go weird. 

**Steps to test this PR**:
1. Use this build: https://staticcdn.duckduckgo.com/macos-desktop-browser/a81f9679-e41d-404d-81a6-fd4a461d82f7/testbuilds/3447618/duckduckgo-1.114.0.305.dmg
1. Test different scenarios (automatic/manual update, trigger the update from different entry points, etc)

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

**Comment by quanganhdo** [comment]
* **#3541** <a href="https://app.graphite.dev/github/pr/duckduckgo/macos-browser/3541?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @quanganhdo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: DuckDuckGo/Updates/UpdateController.swift [hunk 1]
```diff
@@ -71,7 +71,7 @@ final class UpdateController: NSObject, UpdateControllerProtocol {
         didSet {
             if let cachedUpdateResult {
                 latestUpdate = Update(appcastItem: cachedUpdateResult.item, isInstalled: cachedUpdateResult.isInstalled)
-                hasPendingUpdate = latestUpdate?.isInstalled == false && updateProgress.isIdle
+                hasPendingUpdate = latestUpdate?.isInstalled == false && updateProgress.isDone && userDriver?.isResumable == true
```
Comment (tomasstrba) [inline DuckDuckGo/Updates/UpdateController.swift:74]: Just optional: It would help with readability if this is wrapped up to a computed property and documented.

**Review by tomasstrba — APPROVED** [review]
LGTM! ✅ 
Do you think it would make sense to merge to the release branch so we apply this fix one release cycle sooner?
